odoo/addons/test_access_rights/tests/test_res_config.py

```diff
@@ -0,0 +1,30 @@
+from odoo.tests.common import TransactionCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestResConfig(TransactionCase):
+    def test_apply_groups(self):
+        ResGroups = self.env["res.groups"]
+        admin_group = ResGroups.browse(self.ref("base.group_erp_manager"))
+        user_group = ResGroups.browse(self.ref("base.group_user"))
+        no_one_group = ResGroups.browse(self.ref("base.group_no_one"))
+
+        # Sanity check for default status
+        self.assertTrue(no_one_group in user_group.implied_ids)
+        self.assertTrue(no_one_group not in admin_group.implied_ids)
+
+        # Set the setting
+        self.env["res.config.settings"].create([{"group_admin_no_one": True}]).execute()
+
+        # user still implies no_one
+        self.assertTrue(no_one_group in user_group.implied_ids)
+        # admin now _must_ imply no one
+        self.assertTrue(no_one_group in admin_group.implied_ids)
+
+        # Unset the setting
+        self.env["res.config.settings"].create([{"group_admin_no_one": False}]).execute()
+        # admin now _must not_ imply no one
+        self.assertTrue(no_one_group not in admin_group.implied_ids)
+        # user _DOES NOT_ impliy no_one, and that's probably not what we want?
+        # the check below fails:
+        # self.assertTrue(no_one_group in user_group.implied_ids)
```

Comment:
I don't understand what this is saying. The setting which was added implies `group_no_one` on `group_user`, so unsetting it removes the implied group. That applies even if the group was there in the first place, there's no group memory or layers (maybe there should be, mind, but there is not).